docs/tutorial.md

```diff
@@ -0,0 +1,526 @@
+Tutorial
+========
+
+This tutorial will help you ramp up to Circuit with a simple email app.
+
+Note this assumes some prior experience with Compose. See these resources for more information:
+
+- (Android) [Get started with Jetpack Compose](https://developer.android.com/jetpack/compose/documentation)
+- (Android) [Jetpack Compose Tutorial](https://developer.android.com/jetpack/compose/tutorial)
+- (Multiplatform) [Get started with Compose Multiplatform](https://www.jetbrains.com/help/kotlin-multiplatform-dev/compose-multiplatform-getting-started.html)
+
+## Setup
+
+You can do this tutorial in one of two ways:
+
+### 1. Build out of the `tutorial` sample
+
+Clone the circuit repo and work out of the `:samples:tutorial:multiplatform` module. This has all your dependencies set up and ready to go, along with some reusable common code to save you some boilerplate. You can see an implementation of this tutorial there as well. The entry point is `main.kt`. To run the main function, you can run `./gradlew :samples:tutorial:multiplatform:run`.
+
+### 2. Start from scratch
+
+First, set up Compose in your project. See the following guides for more information:
+
+- [Android](https://developer.android.com/jetpack/compose/setup)
+    - Also set up [Parcelize](https://developer.android.com/kotlin/parcelize)
+- [Multiplatform](https://www.jetbrains.com/help/kotlin-multiplatform-dev/compose-multiplatform-getting-started.html)
+
+Next, add the `circuit-foundation` dependency. This includes all the core Circuit artifacts.
+
+=== "build.gradle.kts"
+    ```kotlin
+    dependencies {
+      implementation("com.slack.circuit:circuit-foundation:<version>")
+    }
+    ```
+
+See [setup docs](setup.md) for more information.
+
+## Create a `Screen`
+
+The primary entry points in Circuit are `Screen`s. These are the navigational building blocks of your app. A `Screen` is a simple data class or data object that represents a unique location in your app. For example, a `Screen` could represent an inbox list, an email detail, or a settings screen.
+
+Let's start with a simple `Screen` that represents an inbox list:
+
+=== "Android"
+    ```kotlin
+    @Parcelize
+    data object InboxScreen : Screen
+    ```
+
+=== "Multiplatform"
+    ```kotlin
+    data object InboxScreen : Screen
+    ```
+
+!!! note
+    `Screen` is `Parcelable` on Android. You should use the Parcelize plugin to annotate your screens with `@Parcelize`.
+
+## Design your state
+
+Next, let's define some state for our `InboxScreen`. Circuit uses unidirectional data flow (UDF) to ensure strong separation between presentation logic and UI. States should [_stable_ or _immutable_](https://developer.android.com/jetpack/compose/performance/stability), and directly renderable by your UIs. As such, you should design them to be as simple as possible.
+
+Conventionally, this is written as a nested `State` class inside your `Screen` and _must_ extend `CircuitUiState`.
```

Comment:
Is it? 😅 . Personally I think they're an implementation detail of the presenter + UI.

Reply:
We usually write this structure

```kotlin
data object HomeScreen : Screen {
  data class State(...) : CircuitUiState
  sealed interface Event : CircuitUiEvent { ... }
}
```

Is that what you're referring to? We could add a bit here that state/events could also be their own top-level declarations if they're nontrivial